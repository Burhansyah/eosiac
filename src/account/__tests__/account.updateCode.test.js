const cloneDeep = require(`lodash/cloneDeep`)
const Account = require(`../index`)
const setup = require(`./setup`)

const {env, accountConfig, existingAccount} = setup

describe(`account.updateCode`, () => {
    it(`throws when account does not exist yet`, async () => {
        expect.assertions(1)
        const a = new Account(`test`, accountConfig)
        a.currentState = {}
        expect(a.updateCode({env})).rejects.toThrow(/does not exist/i)
    })

    it(`returns no actions to run when same code and ABI deployed`, async () => {
        expect.assertions(1)
        const a = new Account(`test`, accountConfig)
        a.currentState = cloneDeep(existingAccount)
        a.currentState.ram_quota = accountConfig.ram
        const actions = await a.updateCode({env})
        expect(actions).toBeFalsy()
    })

    it(`returns setcode action on different code hash`, async () => {
        expect.assertions(1)
        const a = new Account(`test`, accountConfig)
        a.currentState = cloneDeep(existingAccount)
        a.currentState.code_hash = `0123456789abcdef`
        const actions = await a.updateCode({env})
        expect(actions).toMatchSnapshot()
    })

    it(`returns setabi action on different abi hash`, async () => {
        expect.assertions(1)
        const a = new Account(`test`, accountConfig)
        a.currentState = cloneDeep(existingAccount)
        a.currentState.abi_hash = `0123456789abcdef`
        const actions = await a.updateCode({env})
        expect(actions).toMatchSnapshot()
    })
})
