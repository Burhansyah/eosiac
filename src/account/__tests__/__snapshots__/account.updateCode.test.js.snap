// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`account.updateCode returns setabi action on different abi hash 1`] = `
Array [
  Object {
    "account": "eosio",
    "authorization": Array [
      Object {
        "actor": "test",
        "permission": "active",
      },
    ],
    "data": Object {
      "abi": "0e656f73696f3a3a6162692f312e31000102686900010475736572046e616d6501000000000000806b026869000000000000",
      "account": "test",
    },
    "name": "setabi",
  },
]
`;

exports[`account.updateCode returns setcode action on different code hash 1`] = `
Array [
  Object {
    "account": "eosio",
    "authorization": Array [
      Object {
        "actor": "test",
        "permission": "active",
      },
    ],
    "data": Object {
      "account": "test",
      "code": "0061736d0100000001370b6000017f60027f7f0060037f7f7f017f60027f7f017f60017e0060017f0060027f7e0060000060037e7e7e0060017f017f60027e7e00029d010903656e7610616374696f6e5f646174615f73697a65000003656e760c656f73696f5f617373657274000103656e76066d656d736574000203656e7610726561645f616374696f6e5f64617461000303656e76066d656d637079000203656e76087072696e74735f6c000103656e76067072696e746e000403656e76067072696e7473000503656e7611656f73696f5f6173736572745f636f6465000603070607080907050a0405017001010105030100010616037f014180c0000b7f0041c1c0000b7f0041c1c0000b070901056170706c79000a0a9a04060400100c0b7a001009200020015104404280808080808080c0eb00200251044020002001100e052000428080808080c0ba98d500520440410042808080d9d3b3ed82ef0010080b0b05428080808080c0ba98d50020015104404280808080aefadeeaa47f2002510440410042818080d9d3b3ed82ef0010080b0b0b4100100d0b800101037f02400240024002402000450d004100410028028c40200041107622016a220236028c404100410028028440220320006a41076a417871220036028440200241107420004d0d0120014000417f460d020c030b41000f0b4100200241016a36028c40200141016a4000417f470d010b41004190c000100120030f0b20030b3601017f230041106b2200410036020c4100200028020c28020041076a417871220036028440410020003602804041003f0036028c400b02000bdb0102047f017e230041106b22022103200224000240024002400240024010002204450d002004418004490d012004100b21020c020b2003420037030841002102200341086a21050c020b20022004410f6a4170716b220224000b2002200410031a20034200370308200341086a2105200441074b0d010b410041bcc00010010b20052002410810041a41a9c000210420032903082106024041a9c0002d00002202450d000240034020024125460d01200441011005200441016a22042d00002202450d020c000b0b20061006200441016a10070b200341106a24000b0b4903004190c0000b2c6661696c656420746f20616c6c6f636174652070616765730048656c6c6f20252066726f6d2068656c6c6f000041bcc0000b0572656164000041000b0448200000",
      "vmtype": 0,
      "vmversion": 0,
    },
    "name": "setcode",
  },
]
`;
